src/category_theory/limits/shapes/zero.lean

```diff
@@ -104,7 +97,8 @@ variables [has_zero_morphisms.{v} C] [has_zero_morphisms.{v'} D]
 begin
   have t : F.functor.map (0 : X ⟶ Y) = F.functor.map (0 : X ⟶ Y) ≫ (0 : F.functor.obj Y ⟶ F.functor.obj Y),
   { apply faithful.injectivity (F.inverse),
-    simp only [functor.map_comp, equivalence.unit, equivalence.unit_inv, equivalence.inv_fun_map, category.assoc],
+    simp only [functor.map_comp],
+    simp,
     dsimp,
     simp, },
```

Comment:
Welcome to dependent type theory, sadly... The basic idea seems to be that, contrary to what you might expect at first, `simp` is not strictly stronger than `dsimp`. While `simp` has access to more facts, sometimes it can't replace an expression when other expressions depend on it. On the other hand, `dsimp` can, because of the guarantee that everything is definitional. Very often in these `simp, dsimp, simp` cycles the dsimp is rewriting an implicit argument (usually some functor acting on an object appearing in the source or target of a morphism), which `simp` can't manage because it would apparently break the type of the morphism, after which things look right syntactically and `simp` can proceed again.

Reply:
So they are unfortunate but I think inevitable. We could write a wrapper tactic to bounce back and forth, but it would probably be unnecessarily slow.

Reply:
In any case I made this proof more explicit, using `rw`.